torchinductor/codegen/triton.py

```diff
@@ -681,9 +681,11 @@ def indexing(
                 mask.append(f"{tree.prefix}mask")
             dense_mask.append(f"{tree.prefix}mask")
 
-        if (need_dense and not have_dense) or index == 0:
+        if (need_dense and not have_dense) or isinstance(
+            index, sympy.core.numbers.Integer
+        ):
             index_str = f"{index_str} + tl.zeros({self.dense_size_str()}, tl.int32)"
-            if index == 0:
+            if isinstance(index, sympy.core.numbers.Integer):
                 return index_str, "None"
```

Comment:
I suppose setting mask to `None` is ok for the example in #1150,`tmp5 = tl.load(in_ptr2 + (1 + tl.zeros([XBLOCK, RBLOCK], tl.int32)), tmp2, other=0)`?

Reply:
Yeah if index is constant we are checking earlier that the constant is valid.